scipy/optimize/zeros.py

```diff
@@ -96,11 +114,20 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
         convenient. If it is None (default), then the normal Newton-Raphson
         or the secant method is used. If it is not None, then Halley's method
         is used.
+    full_output : bool, optional
+        If `full_output` is False, the root is returned.  If `full_output` is
+        True, the return value is ``(x, r)``, where `x` is the root, and `r` is
+        a RootResults object.
+    disp : bool, optional
+        If True, display a RuntimeError if the algorithm didn't converge.
```

Comment:
Not sure if `disp` is the right name for this, because the old behavior is to raise an exception on convergence failure, not only display it. Typically I think `disp` was used to print progress during iterations to stdout, which is not what it means here.

I'd suggest removing the `disp` keyword argument, and retain the error-raising behavior only for `not full_output`. This is how it works e.g. in `fsolve` and elsewhere, so here we probably should follow suit.

Reply:
One reason I added the `disp` keyword was for compatibility with all the other solvers in `zeros.py`: `bisect`, `brentq`, `brenth`, `ridder`.  It is also the case that testing all root-finder methods within a uniform testing framework is much simpler if both the `full_output` and `disp` keywords are present.  

I looked for the actual usages of `disp` and found `call_solver` in `optimize/zeros.c` which sets `PyExc_RuntimeError` with an error message and returns NULL if the solver fails to converge.   I didn't find any progress reporting in these solvers, only reporting of convergence errors.

Are you suggesting that perhaps all the zeros solvers (`brentq/ridder/...`) should match `fsolve` and not have the `disp` keyword?

Reply:
Ah, I did not remember the 1D solvers are inconsistent with fsolve. Then I guess it's best to have this as it is and follow the 1D solver example.